proposals/0012-f3-deals-reliability-ux.md

```diff
@@ -0,0 +1,159 @@
+
+Taking a step back, user-friendly CLI commands are nice, but if we want to push the UX even further, it would be desirable to have a companion graphical web-based deals dashboard that one can point to a node to display the status and details of all deals. An interface like this could be enriched with instructional cues to educate the user progressively.
+
+The Spark team could help prototype this web UI ;-)
+
+#### Impact
+
+_How directly important is the outcome to web3 dev stack product-market fit?_
+
+🔥🔥🔥
+
+This problem statement addresses massive painpoints in the primary use case that Filecoin was built for (file storage and retrieval). Not executing on this translates into first-time user failure and, consequently, bounce.
+
+#### Leverage
+
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+🎯🎯🎯
+
+High. For seeveral reasons: this project will likely land a bunch of refactors that will make the affected areas of the codebase (markets, data transfer, graphsync) more accessible to new developers. Additionally, having a consistent high deal success rate, with a seamless and great UX that gets good user feedback, will get the team much satisfaction, which will translate into higher drive to continue driving improvements. Finally, we hope that new users/devs arriving to Filecoin and having success when making their first deal will translate into interest and hunger for more, and ultimately into some form of contribution.
+
+#### Confidence
+
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+3. We have very solid evidence that the deals flow is... suboptimal. We've used this input to design our action plan. Note that this doc is just a draft and indexes mostly on the problem and not so much in the solution. We'll discover more action items as we execute on the project, but we're confident in the trajectory and themes we're setting.
+
+## Project definition
+
+#### Brief plan of attack
+
+**UX track**
+
+1. Study the user flow in detail. Inventorise the specific elements that are confusing users.
+2. Initiate discussions about user-friendlier terminology.
+3. Design a new CLI UX, prototype it, iterate.
+4. Implement CLI UX changes (and underlying code changes)
+5. One-stop set of JSON-RPC operations that encapsulate the new behaviour with no frills (without leaking implementation details as much as possible).
+
+**Technical reliability track**
+
+0. Do a full inventory of outstanding issues; get to the bottom of the long-standing mystical issues; analyse; fix. Some known tasks:
+1. Testing: Design and implement test suites to reproduce the concrete issues and serve as regression tests. Nightly tests for deals.
```

Comment:
Sounds like @alanshaw was also proposing carving this out

Reply:
We intended this to be local, scoped integration tests that reproduce the memory, datastore, etc. issues -- not distributed, multipartipant, higher-level, acceptance end-to-end tests, which will necesarily exercise many code paths in Lotus.